mesonbuild/interpreter/interpreter.py

```diff
@@ -157,103 +157,17 @@ def dump_value(self, arr, list_sep, indent):
-                                   'native',
-                                   'not_found_message',
-                                   'required',
-                                   'static',
-                                   'version',
-                                   'private_headers',
-                                   'cmake_args',
-                                   'cmake_package_version',
-                                   'include_type',
-                                   },
-                    'declare_dependency': {'include_directories',
-                                           'link_with',
-                                           'sources',
-                                           'dependencies',
-                                           'compile_args',
-                                           'link_args',
-                                           'link_whole',
-                                           'version',
-                                           'variables',
-                                           },
-                    'executable': build.known_exe_kwargs,
-                    'find_program': {'required', 'native', 'version', 'dirs'},
-                    'generator': {'arguments',
-                                  'output',
-                                  'depends',
-                                  'depfile',
-                                  'capture',
-                                  'preserve_path_from'},
-                    'include_directories': {'is_system'},
-                    'install_data': {'install_dir', 'install_mode', 'rename', 'sources'},
-                    'install_headers': {'install_dir', 'install_mode', 'subdir'},
-                    'install_man': {'install_dir', 'install_mode', 'locale'},
-                    'install_subdir': {'exclude_files', 'exclude_directories', 'install_dir', 'install_mode', 'strip_directory'},
-                    'jar': build.known_jar_kwargs,
-                    'project': {'version', 'meson_version', 'default_options', 'license', 'subproject_dir'},
-                    'run_command': {'check', 'capture', 'env'},
-                    'run_target': {'command', 'depends', 'env'},
-                    'shared_library': build.known_shlib_kwargs,
-                    'shared_module': build.known_shmod_kwargs,
-                    'static_library': build.known_stlib_kwargs,
-                    'both_libraries': known_library_kwargs,
-                    'library': known_library_kwargs,
-                    'subdir': {'if_found'},
-                    'subproject': {'version', 'default_options', 'required'},
-                    'test': set.union(_base_test_args, {'is_parallel'}),
-                    'vcs_tag': {'input', 'output', 'fallback', 'command', 'replace_string'},
-                    }
-
+permitted_test_kwargs = {
+    'args', 'depends', 'env', 'should_fail', 'timeout', 'workdir', 'suite',
+    'priority', 'protocol',
+}
```

Comment:
It would be more readable if these were aligned vertically and in alphabetical order.

Reply:
done.